src/connectors/hierarchical-menu/connectHierarchicalMenu.ts

```diff
@@ -81,6 +83,11 @@ export type HierarchicalMenuConnectorParams = {
    * You can also use a sort function that behaves like the standard Javascript [compareFunction](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#Syntax).
    */
   sortBy?: SortBy<HierarchicalMenuItem>;
+  /**
+   * Apply the sorting of facet values defined in settings
+   * Defaults to `true` if sortBy is not given
+   */
+  facetOrdering?: boolean;
```

Comment:
this one is called `facetOrdering` when what it does is actually order the facet values (not just the facets). 
Could this confused with dynamic widgets? do you to name it like this to keep it an implementation detail?

Reply:
The name of the feature is called facet ordering, both for values and for facets. It's a good point that it can be confused, but I'm not sure what the alternative name can be

Reply:
Not directly related to naming.
I understand why e can consider this not to be a breaking change when doing simply`refinementList({ atribute: "brand" })`, as the user has to explicitly set `facetOrdering` in their dashboard for this behaviour to change.

I'm struggling to understand how that's better and more explicit than `refinementList({ sortBy: 'auto' })`.

I understand the following facts:
- `sortBy` wording is slightly confusing for this use case, given there possibly hidden elements via `sortRemaining: "hidden"`, **but so is `facetOrdering`**. 
-  `sortBy` currently has a DEFAULT_SORT we can't change to be `auto` which means users will have to enable it in two places (in the dashboard, and in the widget). But we can make this a default for next majors and use a dev warning in v4. 

What I see is that it's much harder to explain/document:
- the difference between `sortBy` and `facetOrdering` isn't obvious from the outside: both words are two close, 
- the behaviour is tricky: they overlap, they cancel one another (per the test cases, there 12 situations that can determine the behaviour) 
- `sortBy` and `facetOrdering` are here to stay, but they already feel like legacy. 

On the contrary, something like `sortBy: "auto"` is easy to explain, because there are just 3 cases:
   * undefined -> we follow DEFAULT SORTING
   * defined to something like : ["isRefined", "count:asc"] -> follows the order you set
   * define to`"auto"` -> follows the order your set in the dashboard

Reply:
I actually think the right order is:

1. once facet ordering is common, de-emphasise sortBy
2. explain to do all sorting post-ordering in transformItems
3. remove sortBy in a major version

In React InstantSearch for example there's no sortBy, and thus we _need_ a new property `facetOrdering` (https://github.com/algolia/react-instantsearch/pull/3067) to determine to use it or not use it. The logic there is slightly simpler, as it's true by default, as you can't override sortBy.

We first need to write the guides to emphasise not using sortBy in the frontend anymore, as that should be set by facetOrdering, and possibly override in transformItems.

Reply:
My reasoning here (facetOrdering: boolean) is, let's say there's an attribute `brand`. A customer can set the facet ordering on dashboard, and it will be automatically applied to their website (as long as they have up-to-date IS). Let's say they have this widget with `brand` in 5 places across their website. They can disable all these on the dashboard. If they want to disable that facet ordering in only one place, they can apply `facetOrdering: false` to that specific widget.

If we go with `sortBy: 'auto'`, to enable the behavior, they have to go over all the code and put that attribute to 5 places. If they want to disable this at one place, they can modify sortBy to something else. If they want to disable all these facet ordering on `brand`, then they cannot do it on dashboard. They have to modify their code altogether. I feel like this can be a big downside of this approach. After all, this new facet ordering feature, if I understand correctly, is designed to give power to business users so that they can make changes on the dashboard without having to change code every single time they want something.

Reply:
As a final option, we can have the following api:

```
const { sortBy = 'auto' } = widgetParams

// later
getFacetValues(attribute, { sortBy: sortBy === 'auto' ? defaultOrder : sortBy, facetOrdering: sortBy === 'auto'})
```

Some considerations around making it a value for sortBy

1. facetOrdering does more than auto sorting
2. auto doesn't have the reference to facetOrdering, so is confusing alone (name could be different, but not sure what fits, as it's a string)
3. you can't pass a custom fallback sortBy, although that's likely not a huge issue
4. what about React InstantSearch? does it keep a `facetOrdering` prop?
5. using sortBy is one less prop, although it's already very variadic

Reply:
I'd personally suggest we design the interface based on 
* what makes more sense for the user, not how we had to do it in the API side because of internal constraints of ours. Assuming that this might evolve in the futur. I'm referring to the use of a new attribute `facetOrdering` instead of using the existing one facets one and using an `array`
* we keep instant search widgets as loosely coupled to Algolia as possible, so this way it's easier to use in a project that doesn't use Algolia: I'm thinking about people using Algolia with a 3rd party platform - ultimately people who use their service could be potential customer of Algolia tomorrow.

Because of that I personally kite like the approach suggest by @tkrugg of using `sortBy: "auto"`. Wondering if instead of `auto` using `api`, or `auto_from_api` could also make sense this way it's even clearer that it comes from the API.

Reply:
@seafoox 

> I'm referring to the use of a new attribute facetOrdering

The way this PR is proposing, is not having to put `facetOrdering` at all by default. When there are multiple widgets for a same attribute, and you only want to disable facet ordering on a specific widget, then you pass `facetOrdering: false` to the widget. Otherwise, it's `true` by default. Leveraging `sortBy`, customers have to put `sortBy: 'auto'` to every single widget to enable facet ordering whereas with `facetOrdering: true` by default, they don't have to do anything except for configuring on dashboard.

> we keep instant search widgets as loosely coupled to Algolia as possible, so this way it's easier to use in a project that doesn't use Algolia

I don't know if it's what we should pursue. We do not implement features in InstantSearch with other search backend services in mind. If that's what we need to do it, then we need to re-assess our approach, but until then, it's better for us to make things as smooth as possible for Algolia customers.

Reply:
note that we could still default to "auto", if we consider that we don't need to provide a sortBy fallback (it can still be the default_sortby) that can't be overridden, as in my last code snippet

Reply:
We discussed and are going with:

* no facetOrdering option
* facetOrdering is used if you don't provide sortBy (or undefined)

This still gives a default facet ordering, but avoids the two parameters. In React InstantSearch we are keeping facetOrdering as an option, as it has no sortBy to work with